### ISSUE TITLE
Build confidence ellipsoid from NLLOC location files

### ISSUE THREAD

**Issue by megies** [obspy/obspy#1929]
Once #1924 is merged we should also construct `ConfidenceEllipsoid` when reading nonlinloc files.

**Comment by calum-chamberlain** [comment]
This didn't end up being tackled in #1924, but it would still be good to do at some point.

**Comment by megies** [comment]
This would be nice to have @LMurrayBergquist and @DominikStr.

Basically means converting from Azimuth+Dip for Ellipsoid major+minor axis to this major axis Plunge+Azimuth+Rotation notation.

See
 - https://github.com/obspy/obspy/blob/9b18aa8ef604c60927c5586cd4fb956427e5710a/obspy/io/nlloc/tests/data/nlloc.hyp#L10 `EllAz1 331.493 Dip1 -13.2202 Len1 1.37113 Az2 229.814 Dip2 -40.7512 Len2 1.74531`
 - see [QuakeML manual](https://quake.ethz.ch/quakeml/docs/REC?action=AttachFile&do=view&target=QuakeML-BED-20130214b.pdf) page 17
 - see [NonLinLoc file format manual](http://alomax.free.fr/nlloc/soft6.00/formats.html#_location_hypphs_) look for `STATISTICS` line

**Comment by DominikStr** [comment]
I just committed a prototype. After a bit of digging, I found the function needed [here](https://github.com/alomax/utilites/blob/master/matrix_statistics/matrix_statistics.c). There is a reference to a python version I couldn't find. My function is based on the C code. I referenced the author but I don't know if this still conflicts with intellectual property.

As far as I understand it all the transformation should be located as a method `obspy.core.event.base.ConfidenceEllipsoid` ? I will do this as soon as it passes a few tests. The reversed function then shouldn't be a problem.

But at the moment I have trouble finding suitable files of events in both formats to test my code. Any suggestion about where to search?
